File: frontend/src/lib/types/admin-backups.ts
// Mirrors backend/app/schemas/admin_backups.py exactly — keep these two in sync.

export interface BackupPublic {
  id: string;
  started_at: string;
  completed_at: string | null;
  size_bytes: number | null;
  status: string;
  celery_task_id: string | null;
  error_message: string | null;
  ready: boolean;
}

export interface BackupDownloadResponse {
  download_url: string;
}

File: backend/app/schemas/admin_backups.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class BackupPublic(BaseModel):
    id: uuid.UUID
    started_at: datetime
    completed_at: datetime | None
    size_bytes: int | None
    status: str
    celery_task_id: str | None
    error_message: str | None
    ready: bool

    model_config = {"from_attributes": True}


class BackupDownloadResponse(BaseModel):
    download_url: str
